### PR TITLE
Feature/syal 1250

### PR DESCRIPTION
**Pull request by ThanhD-AVISPL**
Hi @MaksimR-AVISPL , this is the code of Sennheiser adapter.
Please help me to review it.
Thanks!

### CODE REVIEW

File: src/main/java/com/avispl/symphony/dal/avdevices/microphone/sennheiser/tcc2/SennheiserTCC2Communicator.java [hunk 1]
```diff
@@ -0,0 +1,846 @@
...
+	private void retrieveDataByCommandName(SennheiserPropertiesList command) {
+		try {
+			byte[] response = send(command.getCommand().getBytes(StandardCharsets.UTF_8));
+			if (response == null || response.length == 0) {
+				updateCachedDeviceData(localCacheMapOfPropertyNameAndValue, command.getName(), SennheiserConstant.NONE);
+				return;
+			}
+			DeviceWrapper deviceWrapper = objectMapper.readValue(response, DeviceWrapper.class);
+			String value = deviceWrapper.getObjectByName(command);
+			updateCachedDeviceData(localCacheMapOfPropertyNameAndValue, command.getName(), value);
```
Comment (MaksimR-AVISPL) [inline src/main/java/com/avispl/symphony/dal/avdevices/microphone/sennheiser/tcc2/SennheiserTCC2Communicator.java:508]: Do you need to check value for null here? 
getObjectByName() may return null

File: src/main/java/com/avispl/symphony/dal/avdevices/microphone/sennheiser/tcc2/SennheiserTCC2Communicator.java [hunk 2]
```diff
@@ -0,0 +1,846 @@
...
+			logger.debug("send data command successfully");
+		}
+	}
+
+	/**
+	 * Constructor set the TCP/IP port to be used as well the default
+	 */
+	public SennheiserTCC2Communicator() {
+		super();
+		this.setPort(45);
```
Comment (MaksimR-AVISPL) [inline src/main/java/com/avispl/symphony/dal/avdevices/microphone/sennheiser/tcc2/SennheiserTCC2Communicator.java]: is port supposed to be fixed? getPort() would provide you one configured in symphony

File: src/main/java/com/avispl/symphony/dal/avdevices/microphone/sennheiser/tcc2/SennheiserTCC2Communicator.java [hunk 3]
```diff
@@ -0,0 +1,846 @@
...
+	@Override
+	public void controlProperties(List<ControllableProperty> controllableProperties) throws Exception {
+		if (CollectionUtils.isEmpty(controllableProperties)) {
+			throw new IllegalArgumentException("ControllableProperties can not be null or empty");
+		}
+		for (ControllableProperty p : controllableProperties) {
+			try {
+				controlProperty(p);
+			} catch (Exception e) {
+				e.printStackTrace();
```
Comment (MaksimR-AVISPL) [inline src/main/java/com/avispl/symphony/dal/avdevices/microphone/sennheiser/tcc2/SennheiserTCC2Communicator.java:361]: please add log entry here with reference to e

File: src/main/java/com/avispl/symphony/dal/avdevices/microphone/sennheiser/tcc2/SennheiserTCC2Communicator.java [hunk 4]
```diff
@@ -0,0 +1,846 @@
...
+					devicesExecutionPool.get(lastIndex).cancel(true);
+				}
+			});
+			try {
+				while (!manageTimeOutWorkerThread.isDone()) {
+					Thread.sleep(100);
+				}
+				manageTimeOutWorkerThread.get();
+			} catch (ExecutionException e) {
+				e.printStackTrace();
```
Comment (MaksimR-AVISPL) [inline src/main/java/com/avispl/symphony/dal/avdevices/microphone/sennheiser/tcc2/SennheiserTCC2Communicator.java:488]: log entry here as well

File: src/main/java/com/avispl/symphony/dal/avdevices/microphone/sennheiser/tcc2/SennheiserTCC2Communicator.java [hunk 5]
```diff
@@ -0,0 +1,846 @@
...
+			} catch (InterruptedException e) {
+				throw new RuntimeException(e);
```
Comment (MaksimR-AVISPL) [inline src/main/java/com/avispl/symphony/dal/avdevices/microphone/sennheiser/tcc2/SennheiserTCC2Communicator.java]: is there any reason to rethrow interrupted exception as Runtime here? Does it require any additional processing?

**Review by MaksimR-AVISPL — APPROVED** [review]

File: src/main/java/com/avispl/symphony/dal/avdevices/microphone/sennheiser/tcc2/SennheiserTCC2Communicator.java (hunk 1 above)
Comment (ThanhD-AVISPL) [reply src/main/java/com/avispl/symphony/dal/avdevices/microphone/sennheiser/tcc2/SennheiserTCC2Communicator.java:508]: i fixed it

**Pull request merged**